src/main/java/AddCommand.java

```diff
@@ -0,0 +1,88 @@
+import java.io.IOException;
+
+/**
+ * AddCommand class handles user's request to add a task to the task list.
+ */
+public class AddCommand extends Command {
+    String type;
+    String description;
+
+    /**
+     * Creates a new AddCommand.
+     * @param type type of task (todo, deadline, event).
+     * @param description description of task.
+     */
+    public AddCommand(String type, String description) {
+        this.type = type;
+        this.description = description;
+    }
+
+    /**
+     * Executes the add command. Creates a task according to the task type and adds it to the task list.
+     * @param tasks list of tasks.
+     * @param ui user interface.
+     * @param storage makeshift database for tasks.
+     * @throws DukeException if user input does not follow input format.
+     * @throws IOException named file exists but is a directory rather than a regular file,
+     *     does not exist but cannot be created, or cannot be open for any other reason.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException, IOException {
+        Task t = new Task(description);
+        switch (type) {
+        case "todo":
+            t = new Todo(description);
+            tasks.add(t);
+            break;
+        case "deadline":
+            checkDateException("deadline");
+            t = new Deadline(description);
+            tasks.add(t);
+            break;
+        case "event":
+            checkDateException("event");
+            t = new Event(description);
+            tasks.add(t);
+            break;
+        default:
+        }
+
+        storage.writeToFile(tasks.saveList());
+        System.out.println("     Got it. I've added this task:");
```

Comment:
Probably better to move the UI stuff to the separate Ui.java as the project guide recommended

Reply:
Thank you! Accidentally missed that out